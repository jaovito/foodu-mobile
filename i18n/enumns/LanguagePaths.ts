export enum LanguagePaths {
  LANDING_ONBOARDING_TITLE = 'onboarding.title',
  LANDING_ONBOARDING_DESCRIPTION = 'onboarding.description',

  SIGNIN_TITLE = 'signIn.title',
  SIGNIN_SIGN_UP = 'signIn.signUp',
  SIGNIN_EMAIL_LABEL = 'signIn.email.label',
  SIGNIN_EMAIL_PLACEHOLDER = 'signIn.email.placeholder',
  SIGNIN_EMAIL_ERROR_REQUIRED = 'signIn.email.errors.required',
  SIGNIN_EMAIL_ERROR_VALID = 'signIn.email.errors.valid',
  SIGNIN_PASSWORD_LABEL = 'signIn.password.label',
  SIGNIN_PASSWORD_PLACEHOLDER = 'signIn.password.placeholder',
  SIGNIN_PASSWORD_ERROR_REQUIRED = 'signIn.password.errors.required',
  SIGNIN_PASSWORD_ERROR_LENGTH = 'signIn.password.errors.length',
  SIGNIN_BUTTON_TEXT = 'signIn.buttonText',

  SIGNUP_TITLE = 'signUp.title',
  SIGNUP_SIGN_IN = 'signUp.alreadyHaveAccount',
  SIGNUP_EMAIL_LABEL = 'signUp.email.label',
  SIGNUP_EMAIL_PLACEHOLDER = 'signUp.email.placeholder',
  SIGNUP_EMAIL_ERROR_REQUIRED = 'signUp.email.errors.required',
  SIGNUP_EMAIL_ERROR_VALID = 'signUp.email.errors.valid',

  SIGNUP_NAME_LABEL = 'signUp.name.label',
  SIGNUP_NAME_PLACEHOLDER = 'signUp.name.placeholder',
  SIGNUP_NAME_ERROR_REQUIRED = 'signUp.name.errors.required',

  SIGNUP_CEL_LABEL = 'signUp.cel.label',
  SIGNUP_CEL_PLACEHOLDER = 'signUp.cel.placeholder',
  SIGNUP_CEL_ERROR_REQUIRED = 'signUp.cel.errors.required',
  SIGNUP_CEL_ERROR_LENGTH = 'signUp.cel.errors.length',

  SIGNUP_PASSWORD_LABEL = 'signUp.password.label',
  SIGNUP_PASSWORD_PLACEHOLDER = 'signUp.password.placeholder',
  SIGNUP_PASSWORD_ERROR_REQUIRED = 'signUp.password.errors.required',
  SIGNUP_PASSWORD_ERROR_LENGTH = 'signUp.password.errors.length',
  SIGNUP_BUTTON_TEXT = 'signUp.buttonText',

  LOCATION_TITLE = 'location.title',
  LOCATION_ADDRESS = 'location.address',
  LOCATION_POSTAL_CODE = 'location.postalCode',
  LOCATION_CITY = 'location.city',
  LOCATION_STATE = 'location.state',
  LOCATION_SAVE = 'location.save',

  LOCATION_ERROR_ADDRESS_REQUIRED = 'location.errors.addressRequired',

  FOOD_SIZE = 'food.size',
  FOOD_ENERGY = 'food.energy',
  FOOD_DELIVERY = 'food.delivery',
  FOOD_ABOUT = 'food.about',
  FOOD_ADD_CART = 'food.addCart',

  GENERAL_YOUR_LOCATION = 'general.yourLocation',
  GENERAL_ORDER_FAVORITE_FOOD = 'general.orderFavoriteFood',
  GENERAL_CITY = 'general.city',
  GENERAL_SEE_MORE = 'general.seeMore',
  GENERAL_NOTHING_TO_SEE = 'general.empty',
  GENERAL_SAVE = 'general.save',
}
