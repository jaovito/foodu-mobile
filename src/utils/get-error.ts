import {AxiosError} from 'axios';

export function getError(err: Error | unknown): AxiosError {
  if (err instanceof AxiosError) {
    return err;
  }

  return err as AxiosError;
}
