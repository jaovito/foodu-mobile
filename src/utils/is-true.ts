/**
 *
 * Convert booleans values, considering the cases that I understand how true and false.
 * Exemple: Empty arrays and objects, we consider false.
 *
 * @param  param - Can be any value
 *
 * @returns if `value` it isn't an empty array, returns true
 * @returns if `value` it isn't an empty object, returns true
 * @returns if `value` it isn't an empty string, returns true
 * @returns if `value` it isn't null, returns true
 * @returns if `value` it's a non-zero number, returns true
 */
export const isTrue = (param: any): boolean => {
  if (typeof param === 'object') {
    if (param === null) {
      return false;
    } else if (param instanceof Array) {
      return param.length === 0 ? false : true;
    } else {
      return Object.entries(param).length === 0 ? false : true;
    }
  }

  return !!param;
};
