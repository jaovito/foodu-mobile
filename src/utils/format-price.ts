import {LanguagePaths} from '@translations/enumns/LanguagePaths';
import {translate} from '@translations/index';

const defaultValues = {
  language: 'en-US',
  currency: 'USD',
};

function getDefaultValues() {
  if (translate(LanguagePaths.GENERAL_LANGUAGE) === 'general.language') {
    return defaultValues;
  }

  return {
    language: translate(LanguagePaths.GENERAL_LANGUAGE),
    currency: translate(LanguagePaths.GENERAL_CURRENCY),
  };
}

export function formatPrice(price: number) {
  return price.toLocaleString(getDefaultValues().language, {
    style: 'currency',
    currency: getDefaultValues().currency,
  });
}
