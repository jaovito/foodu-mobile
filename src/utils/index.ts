import {isTrue} from './is-true';
import {formatPrice} from './format-price';

export {isTrue, formatPrice};
