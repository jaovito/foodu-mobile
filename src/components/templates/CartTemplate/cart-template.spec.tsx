import {mock} from '@test/mocks';
import {fireEvent, render} from '@test/utils/test-utils';
import {LanguagePaths} from '@translations/enumns/LanguagePaths';
import {translate} from '@translations/index';
import {CartTemplate} from './';

describe('<CartTemplate />', () => {
  it('should be defined', () => {
    const cartList = render(
      <CartTemplate
        onSave={mock.cartTemplate.onSave}
        cart={mock.cartTemplate.cart}
        onDelete={mock.cartTemplate.onDelete}
      />,
    );

    expect(cartList).toBeDefined();
  });

  it('should be able to find some food', async () => {
    const {findByText} = render(
      <CartTemplate
        onSave={mock.cartTemplate.onSave}
        cart={mock.cartTemplate.cart}
        onDelete={mock.cartTemplate.onDelete}
      />,
    );

    const cart = await findByText(mock.cartTemplate.cart[1].name);

    expect(cart).toBeTruthy();
  });

  it('should be able to delete some food', async () => {
    const {findAllByTestId} = render(
      <CartTemplate
        onSave={mock.cartTemplate.onSave}
        cart={mock.cartTemplate.cart}
        onDelete={mock.cartTemplate.onDelete}
      />,
    );

    const cart = await findAllByTestId(mock.cartCard.ids.DeleteCartButtonID);
    fireEvent.press(cart[0]);

    expect(mock.cartTemplate.onDelete).toHaveBeenCalled();
  });

  it('should be able to save the cart', async () => {
    const {findByText} = render(
      <CartTemplate
        onSave={mock.cartTemplate.onSave}
        cart={mock.cartTemplate.cart}
        onDelete={mock.cartTemplate.onDelete}
      />,
    );

    const button = await findByText(translate(LanguagePaths.CART_SUBMIT));
    fireEvent.press(button);

    expect(mock.cartTemplate.onSave).toHaveBeenCalled();
  });

  it('should match with snapshot', () => {
    const cartList = render(
      <CartTemplate
        onSave={mock.cartTemplate.onSave}
        cart={mock.cartTemplate.cart}
        onDelete={mock.cartTemplate.onDelete}
      />,
    ).toJSON();

    expect(cartList).toMatchSnapshot();
  });
});
