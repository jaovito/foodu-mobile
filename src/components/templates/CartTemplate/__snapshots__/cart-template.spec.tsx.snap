// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartTemplate /> should match with snapshot 1`] = `
<View
  style={
    [
      {
        "marginTop": 24,
        "paddingBottom": 24,
        "paddingLeft": 24,
        "paddingRight": 24,
        "paddingTop": 24,
      },
    ]
  }
>
  <RCTScrollView
    ListEmptyComponent={
      <TextButton>
        general.empty
      </TextButton>
    }
    contentContainerStyle={{}}
    data={
      [
        {
          "about": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Laboriosam dolor nihil nam blanditiis nobis dignissimos qui itaque, quaerat labore beatae neque esse asperiores, corporis dolore odit quas perferendis, modi explicabo!",
          "counter": 1,
          "created_at": "2023-02-28T17:42:03.934Z",
          "energy": "120 KCal",
          "file": {
            "file_url": "http://",
          },
          "id": "1",
          "name": "Burger",
          "price": 20,
          "size": "BIG",
          "updated_at": "2023-02-28T17:42:03.934Z",
        },
        {
          "about": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Laboriosam dolor nihil nam blanditiis nobis dignissimos qui itaque, quaerat labore beatae neque esse asperiores, corporis dolore odit quas perferendis, modi explicabo!",
          "counter": 2,
          "created_at": "2023-02-28T17:42:03.934Z",
          "energy": "130 KCal",
          "file": {
            "file_url": "http://",
          },
          "id": "2",
          "name": "Pizza",
          "price": 30,
          "size": "BIG",
          "updated_at": "2023-02-28T17:42:03.934Z",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    style={
      [
        {
          "height": "90%",
        },
      ]
    }
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            [
              {
                "alignItems": "flex-start",
                "backgroundColor": "#FFFFFF",
                "borderColor": "rgba(0, 0, 0, 0.04)",
                "borderRadius": 16,
                "borderStyle": "solid",
                "borderWidth": 1,
                "flexDirection": "row",
                "marginBottom": 31,
                "paddingBottom": 24,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 24,
                "shadowColor": "rgba(0, 0, 0, 0.12)",
                "shadowOffset": {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 1,
                "shadowRadius": 8,
                "width": "100%",
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "height": 177,
                  "marginRight": 16,
                  "width": 177,
                },
              ]
            }
          >
            <Image
              onError={[Function]}
              resizeMode="cover"
              source={
                {
                  "uri": "http://",
                }
              }
              style={
                [
                  {
                    "borderRadius": 16,
                    "height": 249,
                    "width": 249,
                  },
                  {
                    "height": "100%",
                    "width": "100%",
                  },
                ]
              }
            />
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "backgroundColor": "red",
                    "borderRadius": 100,
                    "bottom": 16,
                    "height": 47,
                    "justifyContent": "center",
                    "position": "absolute",
                    "right": 16,
                    "width": 47,
                  },
                ]
              }
            >
              <Text
                fontWeight="bold"
                style={
                  [
                    {
                      "color": "#0D0D0D",
                      "fontFamily": "Poppins_700Bold",
                      "fontSize": 31,
                    },
                    {
                      "color": "#FFFFFF",
                      "fontSize": 24,
                      "lineHeight": 29,
                    },
                  ]
                }
              >
                1
              </Text>
            </View>
          </View>
          <View
            style={
              [
                {
                  "alignItems": "flex-start",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "height": "90%",
                  "justifyContent": "space-between",
                },
              ]
            }
          >
            <Text
              fontWeight="bold"
              style={
                [
                  {
                    "color": "#0D0D0D",
                    "fontFamily": "Poppins_700Bold",
                    "fontSize": 31,
                  },
                  {
                    "fontSize": 35,
                  },
                ]
              }
            >
              Burger
            </Text>
            <Text
              fontWeight="regular"
              numberOfLines={3}
              style={
                [
                  {
                    "color": "#0D0D0D",
                    "fontFamily": "Poppins_400Regular",
                    "fontSize": 31,
                  },
                  {
                    "fontSize": 24,
                    "maxWidth": "100%",
                  },
                ]
              }
            >
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Laboriosam dolor nihil nam blanditiis nobis dignissimos qui itaque, quaerat labore beatae neque esse asperiores, corporis dolore odit quas perferendis, modi explicabo!
            </Text>
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "gap": 16,
                    "justifyContent": "flex-end",
                    "marginTop": 16,
                    "width": "100%",
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#0D0D0D",
                      "fontFamily": "Poppins_400Regular",
                      "fontSize": 31,
                    },
                    {},
                  ]
                }
              >
                $
                20.00
              </Text>
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "gap": 8,
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <RNGestureHandlerButton
                  activeOpacity={0.8}
                  borderless={true}
                  collapsable={false}
                  delayLongPress={600}
                  handlerTag={29}
                  handlerType="NativeViewGestureHandler"
                  onActiveStateChange={[Function]}
                  onGestureEvent={[Function]}
                  onGestureHandlerEvent={[Function]}
                  onGestureHandlerStateChange={[Function]}
                  onHandlerStateChange={[Function]}
                  onPress={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#FE8C00",
                      "borderRadius": 31,
                      "height": 63,
                      "justifyContent": "center",
                      "opacity": 1,
                      "width": 63,
                    }
                  }
                  testID="DecrementCounterButtonID"
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      [
                        {
                          "color": undefined,
                          "fontSize": 12,
                        },
                        [
                          {
                            "color": "#FFFFFF",
                            "fontSize": 31,
                          },
                        ],
                        {
                          "fontFamily": "Feather",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    ï†£
                  </Text>
                </RNGestureHandlerButton>
                <RNGestureHandlerButton
                  activeOpacity={0.8}
                  borderless={true}
                  collapsable={false}
                  delayLongPress={600}
                  handlerTag={30}
                  handlerType="NativeViewGestureHandler"
                  onActiveStateChange={[Function]}
                  onGestureEvent={[Function]}
                  onGestureHandlerEvent={[Function]}
                  onGestureHandlerStateChange={[Function]}
                  onHandlerStateChange={[Function]}
                  onPress={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#FE8C00",
                      "borderRadius": 31,
                      "height": 63,
                      "justifyContent": "center",
                      "opacity": 1,
                      "width": 63,
                    }
                  }
                  testID="IncrementCounterButtonID"
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      [
                        {
                          "color": undefined,
                          "fontSize": 12,
                        },
                        [
                          {
                            "color": "#FFFFFF",
                            "fontSize": 31,
                          },
                        ],
                        {
                          "fontFamily": "Feather",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    ï‡€
                  </Text>
                </RNGestureHandlerButton>
                <RNGestureHandlerButton
                  activeOpacity={0.8}
                  borderless={true}
                  collapsable={false}
                  color="red"
                  delayLongPress={600}
                  handlerTag={31}
                  handlerType="NativeViewGestureHandler"
                  onActiveStateChange={[Function]}
                  onGestureEvent={[Function]}
                  onGestureHandlerEvent={[Function]}
                  onGestureHandlerStateChange={[Function]}
                  onHandlerStateChange={[Function]}
                  onPress={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "red",
                      "borderRadius": 31,
                      "height": 63,
                      "justifyContent": "center",
                      "opacity": 1,
                      "width": 63,
                    }
                  }
                  testID="DeleteCartButtonID"
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      [
                        {
                          "color": undefined,
                          "fontSize": 12,
                        },
                        [
                          {
                            "color": "#FFFFFF",
                            "fontSize": 31,
                          },
                        ],
                        {
                          "fontFamily": "Feather",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    ï‡´
                  </Text>
                </RNGestureHandlerButton>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            [
              {
                "alignItems": "flex-start",
                "backgroundColor": "#FFFFFF",
                "borderColor": "rgba(0, 0, 0, 0.04)",
                "borderRadius": 16,
                "borderStyle": "solid",
                "borderWidth": 1,
                "flexDirection": "row",
                "marginBottom": 31,
                "paddingBottom": 24,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 24,
                "shadowColor": "rgba(0, 0, 0, 0.12)",
                "shadowOffset": {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 1,
                "shadowRadius": 8,
                "width": "100%",
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "height": 177,
                  "marginRight": 16,
                  "width": 177,
                },
              ]
            }
          >
            <Image
              onError={[Function]}
              resizeMode="cover"
              source={
                {
                  "uri": "http://",
                }
              }
              style={
                [
                  {
                    "borderRadius": 16,
                    "height": 249,
                    "width": 249,
                  },
                  {
                    "height": "100%",
                    "width": "100%",
                  },
                ]
              }
            />
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "backgroundColor": "red",
                    "borderRadius": 100,
                    "bottom": 16,
                    "height": 47,
                    "justifyContent": "center",
                    "position": "absolute",
                    "right": 16,
                    "width": 47,
                  },
                ]
              }
            >
              <Text
                fontWeight="bold"
                style={
                  [
                    {
                      "color": "#0D0D0D",
                      "fontFamily": "Poppins_700Bold",
                      "fontSize": 31,
                    },
                    {
                      "color": "#FFFFFF",
                      "fontSize": 24,
                      "lineHeight": 29,
                    },
                  ]
                }
              >
                2
              </Text>
            </View>
          </View>
          <View
            style={
              [
                {
                  "alignItems": "flex-start",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "height": "90%",
                  "justifyContent": "space-between",
                },
              ]
            }
          >
            <Text
              fontWeight="bold"
              style={
                [
                  {
                    "color": "#0D0D0D",
                    "fontFamily": "Poppins_700Bold",
                    "fontSize": 31,
                  },
                  {
                    "fontSize": 35,
                  },
                ]
              }
            >
              Pizza
            </Text>
            <Text
              fontWeight="regular"
              numberOfLines={3}
              style={
                [
                  {
                    "color": "#0D0D0D",
                    "fontFamily": "Poppins_400Regular",
                    "fontSize": 31,
                  },
                  {
                    "fontSize": 24,
                    "maxWidth": "100%",
                  },
                ]
              }
            >
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Laboriosam dolor nihil nam blanditiis nobis dignissimos qui itaque, quaerat labore beatae neque esse asperiores, corporis dolore odit quas perferendis, modi explicabo!
            </Text>
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "gap": 16,
                    "justifyContent": "flex-end",
                    "marginTop": 16,
                    "width": "100%",
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#0D0D0D",
                      "fontFamily": "Poppins_400Regular",
                      "fontSize": 31,
                    },
                    {},
                  ]
                }
              >
                $
                60.00
              </Text>
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "gap": 8,
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <RNGestureHandlerButton
                  activeOpacity={0.8}
                  borderless={true}
                  collapsable={false}
                  delayLongPress={600}
                  handlerTag={32}
                  handlerType="NativeViewGestureHandler"
                  onActiveStateChange={[Function]}
                  onGestureEvent={[Function]}
                  onGestureHandlerEvent={[Function]}
                  onGestureHandlerStateChange={[Function]}
                  onHandlerStateChange={[Function]}
                  onPress={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#FE8C00",
                      "borderRadius": 31,
                      "height": 63,
                      "justifyContent": "center",
                      "opacity": 1,
                      "width": 63,
                    }
                  }
                  testID="DecrementCounterButtonID"
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      [
                        {
                          "color": undefined,
                          "fontSize": 12,
                        },
                        [
                          {
                            "color": "#FFFFFF",
                            "fontSize": 31,
                          },
                        ],
                        {
                          "fontFamily": "Feather",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    ï†£
                  </Text>
                </RNGestureHandlerButton>
                <RNGestureHandlerButton
                  activeOpacity={0.8}
                  borderless={true}
                  collapsable={false}
                  delayLongPress={600}
                  handlerTag={33}
                  handlerType="NativeViewGestureHandler"
                  onActiveStateChange={[Function]}
                  onGestureEvent={[Function]}
                  onGestureHandlerEvent={[Function]}
                  onGestureHandlerStateChange={[Function]}
                  onHandlerStateChange={[Function]}
                  onPress={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#FE8C00",
                      "borderRadius": 31,
                      "height": 63,
                      "justifyContent": "center",
                      "opacity": 1,
                      "width": 63,
                    }
                  }
                  testID="IncrementCounterButtonID"
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      [
                        {
                          "color": undefined,
                          "fontSize": 12,
                        },
                        [
                          {
                            "color": "#FFFFFF",
                            "fontSize": 31,
                          },
                        ],
                        {
                          "fontFamily": "Feather",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    ï‡€
                  </Text>
                </RNGestureHandlerButton>
                <RNGestureHandlerButton
                  activeOpacity={0.8}
                  borderless={true}
                  collapsable={false}
                  color="red"
                  delayLongPress={600}
                  handlerTag={34}
                  handlerType="NativeViewGestureHandler"
                  onActiveStateChange={[Function]}
                  onGestureEvent={[Function]}
                  onGestureHandlerEvent={[Function]}
                  onGestureHandlerStateChange={[Function]}
                  onHandlerStateChange={[Function]}
                  onPress={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "red",
                      "borderRadius": 31,
                      "height": 63,
                      "justifyContent": "center",
                      "opacity": 1,
                      "width": 63,
                    }
                  }
                  testID="DeleteCartButtonID"
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      [
                        {
                          "color": undefined,
                          "fontSize": 12,
                        },
                        [
                          {
                            "color": "#FFFFFF",
                            "fontSize": 31,
                          },
                        ],
                        {
                          "fontFamily": "Feather",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    ï‡´
                  </Text>
                </RNGestureHandlerButton>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <RNGestureHandlerButton
    activeOpacity={0.105}
    collapsable={false}
    delayLongPress={600}
    handlerTag={35}
    handlerType="NativeViewGestureHandler"
    onActiveStateChange={[Function]}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    onPress={
      [MockFunction] {
        "calls": [
          [],
        ],
        "results": [
          {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FE8C00",
        "borderRadius": 16,
        "justifyContent": "center",
        "marginTop": 16,
        "paddingBottom": 31,
        "paddingLeft": 31,
        "paddingRight": 31,
        "paddingTop": 31,
        "width": "100%",
      }
    }
    testID="ButtonTestID"
    underlayColor="black"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "black",
          "borderBottomLeftRadius": undefined,
          "borderBottomRightRadius": undefined,
          "borderRadius": 16,
          "borderTopLeftRadius": undefined,
          "borderTopRightRadius": undefined,
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <Text
      style={
        [
          {
            "color": "#0D0D0D",
            "fontFamily": "Poppins_400Regular",
            "fontSize": 31,
          },
          {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      cart.submit
    </Text>
  </RNGestureHandlerButton>
</View>
`;
